src/services/coingecko/api/price.service.ts

```diff
@@ -189,12 +189,12 @@ export class PriceService {
           for (const key in result) {
             const value = result[key];
             const [timestamp, price] = value;
-            if (timestamp > dayTimestamp * 1000) {
-              prices[dayTimestamp * 1000] = price;
-              dayTimestamp += twentyFourHoursInSecs;
+            if (timestamp > start * 1000) {
```

Comment:
Are the timestamps always divisable by 86400 / 3600? As it looks like that's what the dayTimestamp variable was doing previously, making sure that the prices array had an entry every 86400.